c/compare.sh

```diff
@@ -71,14 +71,10 @@ for f in $SETS ; do
   for ff in $(ls $(grep -v "^#" $f)) ; do
     orig=$ff
 
-    # Get file that belongs to task definition (by file name)
-    # It would be more precise, but also a lot more complicated,
-    # to use the input-file value of the task definition.
     if echo $ff | grep -q ".yml$"; then
-      ff=$(echo $ff | sed 's/\.yml$/.i/')
-      if [ ! -f $ff ]; then
-        ff=$(echo $ff | sed 's/\.i$/.c/')
-      fi
+      # The cut removes quotes around the file name
+      input_basename=$(cat $ff | yq '.input_files' | cut -d'"' -f 2)
```

Comment:
`jq` has `--raw-output` for outputting strings without quotes. And you have a [useless use of cat](https://en.wikipedia.org/wiki/Cat_(Unix)#Useless_use_of_cat) here, `yq` can read files directly.

And while we are at it, please add quotes to keep the script working even with whitespace etc. in file names (`input_basename="$(...)"` and similarly in the next line). It is not probable that such files will be added, but it is a good habit to always add quotes.

Reply:
> jq has --raw-output for outputting strings without quotes. And you have a useless use of cat here, yq can read files directly.

Thanks, this is fixed.

> And while we are at it, please add quotes to keep the script working even with whitespace etc. in file names (input_basename="$(...)" and similarly in the next line). It is not probable that such files will be added, but it is a good habit to always add quotes.

Are you sure this is necessary? The command substitution should return a single string. For the use of `$ff`, quotes are necessary in case it contains whitespace, so I've added them in the new commit, but adding quotes around the substitutions seems unnecessary.

Reply:
Quotes around the substitution are necessary if whitespace is contained. So, as I said, this is not probable here, but it is still a good habit.